src/main/java/cz/cuni/lf1/lge/ThunderSTORM/calibration/PhasorAbstractCalibrationProcess.java

```diff
@@ -0,0 +1,547 @@
+        // fit a quadratic polynomial to sigma1 = f(zpos) and sigma1 = f(zpos) for each bead
+        IterativeFitting polynomialFitter = new IterativeFitting(config.inlierFittingMaxIters, config.inlierFittingInlierFraction);
+        allPolynomsS1 = new ArrayList<DefocusFunction>();
+        allPolynomsS2 = new ArrayList<DefocusFunction>();
+
+        List<double[]> framesArrays = new ArrayList<double[]>();
+        List<double[]> sigma1Arrays = new ArrayList<double[]>();
+        List<double[]> sigma2Arrays = new ArrayList<double[]>();
+
+                
+        AtomicInteger moleculesProcessed = new AtomicInteger(0);
+        usedPositions = new ArrayList<PSFSeparator.Position>();
+        for(PSFSeparator.Position p : positions) {
+            moleculesProcessed.incrementAndGet();
+            IJ.showProgress(0.9 + 0.1 * (double) moleculesProcessed.intValue() / (double) positions.size());
+            IJ.showStatus("Fitting polynoms: molecule " + moleculesProcessed + " of " + positions.size() + "...");
+
+            try {
+                //if(p.getSize() < config.minimumFitsCount) {
+                //    continue;
+                //}else{
+            //IJ.log("Not enough fits to continue: "+p.getSize());}
+                
+                double z0guess = guessZ0(imp,p);
+                p.discardFitsByFrameRange(z0guess - zRange/stageStep, z0guess + zRange/stageStep);
+
+                // retrieve values again after filtering out fits not in range
+                double[] framesArrayOfZ = p.getFramesAsArrayOfZ(z0guess, stageStep);
+                double[] sigma1Array = p.getAsArray(LABEL_SIGMA1);
+                double[] sigma2Array = p.getAsArray(LABEL_SIGMA2);
+                // fit s1,2 = polynomial(frame)
+                DefocusFunction polynomS1;
+                DefocusFunction polynomS2;
+                //int polynomS2;
+                try {
+                    polynomS1 = polynomialFitter.fitParams(defocusModel, framesArrayOfZ, sigma1Array, config.polyFitMaxIters);
+                    polynomS2 = polynomialFitter.fitParams(defocusModel, framesArrayOfZ, sigma2Array, config.polyFitMaxIters);
+                } catch(TooManyEvaluationsException e) {
+                    //discard not converged
+                    //IJ.log(e.toString());
+                    continue;
+                } catch(ArrayIndexOutOfBoundsException ex) {
+                    // discard: no detections!
+                    continue;
+                }
+
+                // defocus out of range?
+                if(config.checkIfDefocusIsInRange && (!isInZRange(polynomS1.getC()))) {
+                    continue;
+                }
+                // find the center point between the minima of the two polynomials and shift the origin
+                //double intersection = (polynomS1.getC() + polynomS2.getC()) / 2;
+                double intersection = (polynomS1.getC() + 1) / 2;
```

Comment:
I don't get this change.

Reply:
Thanks for your work.

This change can be removed - the correct line should be 
double intersection = (polynomS1.getC() + polynomS2.getC()) / 2;

For the rest I'll number them as I go: 1) as far as I'm aware, you can commit to this pull request (Allow edits from maintainers is turned on).

I think there are currently some errors when trying to get calibration curves:
2) During testing, I got an error during 3D cylindrical lens calibration with both gaussian and phasor:
java.lang.NullPointerException
	at cz.cuni.lf1.lge.ThunderSTORM.calibration.AbstractCalibrationProcess.guessZ0(AbstractCalibrationProcess.java:333)
	at cz.cuni.lf1.lge.ThunderSTORM.calibration.AbstractCalibrationProcess.fitQuadraticPolynomials(AbstractCalibrationProcess.java:149)
	at cz.cuni.lf1.lge.ThunderSTORM.calibration.PhasorAstigmaticCalibrationProcess.runCalibration(PhasorAstigmaticCalibrationProcess.java:33)
	at cz.cuni.lf1.lge.ThunderSTORM.CylindricalLensCalibrationPlugin.run(CylindricalLensCalibrationPlugin.java:90)

This can be fixed by changing in lines 332-336 all instances of 'intensityAsArray' to 'sigma1AsArray'. The reason is that phasor does not provide intensity, thus the intensityAsArray is length 0, giving problems when trying to loop over it.

3) However, with that fixed, astigmatic calibration ALWAYS uses phasor to measure sigma1 and sigma2. I believe this was the reason I split AbstractCalibrationProcess.java in 2 files (one for phasor).

4) Next, a small non-necessary addition that I would like, considering the plot ThunderSTORM creates when calibration is performed. Currently (AbstractCalibrationProcess, line 501), you set Y limits from 0 to stageStep, which works fine for gaussian fitting. For phasor, this is wildly over-setting the Y limits - I used to set them to 0.9 to 1.0. Is it possible/easy for you to set them 0.9-1.0 for phasor only? 

5) Currently, there is a problem with setting the crossing of the calibration curves to zero. In fact, this was the standard option before I started adding phasor, but when I compile the sept. 10, 2016 version, the calibration curves do not center at zero:

![image](https://user-images.githubusercontent.com/32260034/31187731-0cad542c-a933-11e7-9032-999b8a21c2d3.png)

This is working correctly (same file, same settings), with ThunderSTORM v1.3, but a lot changed between them. Could you take a look at it?

Thanks,